### PR TITLE
Fixes #1695

### LINKED ISSUE #1695: [MSFT_TeamsTenantDialPlan] Does not export multiple Normalization rules
#### Details of the scenario you tried and the problem that is occurring
I tried exporting a tenant dial plan for Teams which has multiple normalization rules. Only the first rule is being exported, the rest is being ommitted.
#### Suggested solution to the issue
I believe the issue is in https://github.com/microsoft/Microsoft365DSC/blob/ccb52e536b12c7a28285bf24bb59504928e0fbc8/Modules/Microsoft365DSC/DSCResources/MSFT_TeamsTenantDialPlan/MSFT_TeamsTenantDialPlan.psm1#L442

I have checked the type of the return value for normalization rules:
```
$p = Get-CsTenantDialPlan
$p[0].NormalizationRules.gettype()


IsPublic IsSerial Name                                     BaseType
-------- -------- ----                                     --------
True     True     ArrayList                                System.Object
```

The code is checking for a hashtable. Thus, it defaults to the else section where it only picks the first rule. The type needs to be corrected.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
# insert configuration here
```

### PR DESCRIPTION
**Pull request by NikCharlebois**
#### Pull Request (PR) description
*TeamsTenantDialPlan
  * Fixed an issue where the Export only extracted the first
    normalization rule.

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

**Comment by NikCharlebois** [comment]
@ykuijs can you please do the code review? Thanks

File: Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1 [hunk 1]
```diff
@@ -77,7 +77,7 @@ function Get-TeamByName
     $loopCounter = 0
     do
     {
-        $team = Get-Team -DisplayName $TeamName
+        $team = Get-Team -DisplayName $TeamName | Where-Object -FilterScript {$_.DisplayName -eq $TeamName}
```
Comment (ykuijs) [inline Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1:80]: Parameters look the same, both checking DisplayName against $TeamName. Shouldn't it be either the parameter or the Where-Object?

**Review by ykuijs — CHANGES_REQUESTED** [review]
One question

File: Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1 (hunk 1 above)
Comment (NikCharlebois) [reply Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1:80]: Get-Team -DisplayName, based on documentation is a wild card search. In our case we want an exact match:
https://docs.microsoft.com/en-us/powershell/module/teams/get-team?view=teams-ps#example-4

**Review by ykuijs — APPROVED** [review]
LGTM

**Pull request merged**